service/history/transferQueueStandbyTaskExecutor.go

```diff
@@ -270,16 +273,53 @@ func (t *transferQueueStandbyTaskExecutor) processCloseExecution(
 			workflowHistoryLength,
 			visibilityMemo,
 			searchAttr,
-		)
+		); err != nil {
+			return nil, err
+		}
+
+		// verify if parent got the completion event
+		verifyCompletionRecorded := mutableState.HasParentExecution() && executionInfo.NewExecutionRunId == ""
+		if verifyCompletionRecorded {
+			_, err := t.historyClient.VerifyChildExecutionCompletionRecorded(ctx, &historyservice.VerifyChildExecutionCompletionRecordedRequest{
+				NamespaceId: executionInfo.ParentNamespaceId,
+				WorkflowExecution: &commonpb.WorkflowExecution{
+					WorkflowId: executionInfo.ParentWorkflowId,
+					RunId:      executionInfo.ParentRunId,
+				},
+				CompletedExecution: &commonpb.WorkflowExecution{
+					WorkflowId: transferTask.WorkflowID,
+					RunId:      transferTask.RunID,
+				},
+				ParentInitiatedId:      executionInfo.ParentInitiatedId,
+				ParentInitiatedVersion: executionInfo.ParentInitiatedVersion,
+				Clock:                  executionInfo.ParentClock,
+			})
+			switch err.(type) {
+			case nil, *serviceerror.NotFound, *serviceerror.NamespaceNotFound:
+				return nil, nil
+			case *serviceerrors.WorkflowNotReady:
+				return &verifyChildCompletionRecordedInfo{}, nil
```

Comment:
How is verifyChildCompletionRecordedInfo used? I thought we should return WorkflowNotReady?

Reply:
It's never used and the idea is to return a non-nil pointer so that the post action func can be executed. 

The return value for this action func is (someInfo, error). 
- If returned error is not nil, then task is considered failure and postActionFunc (which based on how long the task has been executed can be no-op or drop) will not be called. 
- If error nil, someInfo is not nil. It means, no error but verification fails. Post action func will be triggered and return either ErrTaskRetry or ErrTaskDiscard based on task processing time.
- If both are nil, verification succeeded, task can be acked.

So here, it's the second case and we need to return some non-nil info. I can actually do `return &struct{}{}, nil` but it looks a bit weird to me. So created a named empty struct in case we need other info later. 

I can add some comments here.

Reply:
I see, so the someInfo will be used by the postActionFn, but in this case it is standbyTaskPostActionNoOp, which does not need any extra field. That make sense. Please do add some simple comments.

Reply:
Or you can define a const as:
const verifyChildCompletionRecordedInfo = &struct{}{}
and put comments about why we need such empty struct.